cmd/govim/quickfix.go

```diff
@@ -75,20 +76,46 @@ func (v *vimstate) updateQuickfixWithDiagnostics(force bool, wasPrevNotDiagnosti
 	if !wasPrevNotDiagnostics && len(v.lastQuickFixDiagnostics) > 0 {
 		var want qflistWant
 		v.Parse(v.ChannelExpr(`getqflist({"idx":0})`), &want)
-		if want.Idx == 0 {
+		if want.Idx == 1 {
+			// Don't guess selection if position is 1
 			goto NewIndexSet
 		}
 		wantIdx := want.Idx - 1
 		if len(v.lastQuickFixDiagnostics) <= wantIdx {
```

Comment:
I really can't remember what condition this is catching. @tbruyelle do you have any intuition? If so, please can you add a comment? Otherwise, I think we should drop this conditional and associated block

Reply:
At first look, I thought it was there to skip index selection when the current index is higher than the number of new fixes, but actually it doesn't compare with the new list length but with the previous list length...

I don't know if this situation can happen actually. I put a panic inside the `if` and run the tests, no errors, so I think we probably can remove it :)

Reply:
Let's drop it and see what breaks :) Easy to then add a fix back along with a comment!

Reply:
done f7c8ec1